### PR TITLE
HIP-584:  Add `MintLogic` + `BurnLogic`

### LINKED ISSUE #6235: HIP-584: Copy `MintLogic` into `hedera-mirror-web3`
### Problem

In order to support eth_estimateGas for `MintPrecompile` we should first adapt and copy `MintLogic`.

### Solution

Copy `MintLogic` in `com.hedera.services` package

### Alternatives

_No response_

### LINKED ISSUE #6236: HIP-584: Copy `BurnLogic` into `hedera-mirror-web3`
### Problem

In order to support `eth_estimateGas` for `BurnPrecompile` we should first adapt and copy `BurnLogic`.

### Solution

Copy `BurnLogic` in `com.hedera.services` package

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:

**Related issue(s)**:

Fixes #6235 
Fixes #6236

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/BurnLogic.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ *
+ * Differences with the original:
+ *  1. Removed validations performed in UsageLimits, since they check global node limits,
+ *  while on Archive Node we are interested in transaction scope only
+ *  2. Use abstraction for the state by introducing {@link Store} interface
+ *  3. Use copied models from hedera-services which are enhanced with additional constructors for easier setup,
+ *  those are {@link Account}, {@link Token}, {@link TokenRelationship}
+ *  4. validateSyntax method uses default value of true for areNftsEnabled property
+ * */
+public class BurnLogic {
```
Comment (0xivanov) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/BurnLogic.java:55]: Shouldn't we make the logic types beans?

**Review by 0xivanov — COMMENTED** [review]
Looks good

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java [hunk 2]
```diff
@@ -65,6 +65,12 @@ public class MirrorNodeEvmProperties implements EvmProperties {
     @Getter
     private long htsDefaultGasCost = 10000;
 
+    @Getter
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:68]: Should put `@Min(1)` on all of these new properties.

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/BurnLogic.java (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/BurnLogic.java:55]: Can create them as beans inside a Spring configuration class

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java (hunk 2 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:68]: Resolved

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/BurnLogic.java (hunk 1 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/BurnLogic.java:55]: Added beans configuration

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**